mobile/src/main/java/com/truckmuncher/truckmuncher/vendor/menuadmin/ResetVendorTrucksService.java

```diff
@@ -0,0 +1,37 @@
+package com.truckmuncher.truckmuncher.vendor.menuadmin;
+
+import android.app.IntentService;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Parcelable;
+
+import com.truckmuncher.truckmuncher.data.Contract;
+
+public class ResetVendorTrucksService extends IntentService {
+
+    private static final String ARG_VALUES = "content_values";
+
+    public ResetVendorTrucksService() {
+        super(ResetVendorTrucksService.class.getSimpleName());
+    }
+
+    public static Intent newIntent(Context context, ContentValues[] contentValues) {
+        Intent intent = new Intent(context, ResetVendorTrucksService.class);
+        intent.putExtra(ARG_VALUES, contentValues);
+        return intent;
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        Parcelable[] parcelables = intent.getParcelableArrayExtra(ARG_VALUES);
+        ContentValues[] contentValues = new ContentValues[parcelables.length];
+
+        for (int i = 0; i < parcelables.length; i++) {
+            contentValues[i] = (ContentValues) parcelables[i];
+        }
+
+        getContentResolver().bulkInsert(Contract.TRUCK_PROPERTIES_URI, contentValues);
```

Comment:
Instead of a bulk insert, do a batch of updates. That way, you don't need to pull all the data out of the db and put it all in again. 

`getContentResolver().applyBatch()`. There's a pretty good write up available about it here http://www.grokkingandroid.com/better-performance-with-contentprovideroperation/

Then the `IntentService` here could even just take a list/array of ids for the db instead of full `ContentValues`